### DISCUSSION TITLE
Accessing API with multiple embedded presentations

### DISCUSSION THREAD

**Discussion by jaybonthius** [hakimel/reveal.js#3126] [Q&A]
In the documentation for [API Methods](https://revealjs.com/api/), it says

> If you're working with a single presentation instance the API can be accessed via the global `Reveal` object.

But what if I'm working with _multiple_ embedded presentations? I've tried a few of different things, none of which work.

On my page, I have a few embedded presentations which I initialize in a `for` loop. Then I try to detect `slidechanged` events with `.on('slidechanged', ...)`. This correctly detects slide changes on each presentation, but then `console.log(rev.getCurrentSlide())` outputs the current slide of the _last presentation on the page_ (regardless of which presentation had a `slidechanged` event).

```js
var elms = document.getElementsByClassName( 'reveal' );
for ( var i = 0; i < elms.length; i++ ) {
    var rev = new Reveal( elms[ i ], {
        // config options
    } );

    rev.initialize();
    
    rev.on( 'slidechanged', event => { 
        console.log(rev.getCurrentSlide());
    });
}
```

What am I doing wrong?

**Comment by hakimel** [comment]
You can fix this by changing `var rev` to `let rev`. 

The error here is due to how hoisting works in JS. Your `var` definition inside the for loop is hoisted to the scope/function where the for loop runs. This means your event listener is always referring to the same `rev` variable.

Using `let` or `const` makes it work because those variable types are only hoisted to the top of the current block (the for loop in this case).
